src/BaGetter.Aliyun/AliyunApplicationExtensions.cs

```diff
@@ -10,8 +12,14 @@ namespace BaGetter;
 
 public static class AliyunApplicationExtensions
 {
-    public static BaGetterApplication AddAliyunOssStorage(this BaGetterApplication app)
+    private const string AliyunOss = "AliyunOss";
+
+    public static BaGetterApplication AddAliyunOssStorage(this BaGetterApplication app, IConfiguration configuration)
     {
+        if (!configuration.HasStorageType(AliyunOss)) return app;
+
+        StatisticsHelperUsedServices.AddServiceToServices(AliyunOss);
```

Comment:
Instead of doing this, could the statistics service load the `StorageOptions` and look at its `Type` property?

Reply:
Agreed, just have the service inject `IOptions<BaGetterOptions>` and read the configured types from there. That's less intrusive to the services and configuration calls.

Reply:
The current system allows a configuration change during runtime, without restarting BaGetter, right? I'd like to keep that feature, I think it's nice to be able to change e.g. the search service with zero downtime. Not injecting the service in the first place would break this and force a restart of BaGetter in order to inject the newly configured services.